src/ocl_util.cc

```diff
@@ -226,11 +246,18 @@ __constant uint32_t const U32_MAX = 0xffffffff;
       assert_st( !init_done.v );
       cl_get_platforms( platforms );
       if( platforms.empty() ) { rt_err( "no OpenCL platforms found" ); }
-      for( vect_cl_platform_id::const_iterator i = platforms.begin(); i != platforms.end(); ++i ) {
-	vect_cl_device_id devices;
-	cl_get_devices( devices, *i, CL_DEVICE_TYPE_GPU );
-	if( !devices.empty() ) { use_devices = vect_cl_device_id{devices[0]}; } // pick first device only (arbitrarily)
+      if( use_platform_id >= platforms.size() ) { rt_err( "desired platform ID is not available" ); }
+      vect_cl_device_id devices;
+      cl_get_devices( devices, platforms[use_platform_id], CL_DEVICE_TYPE_GPU );
+      if( !devices.empty() ) {
```

Comment:
i don't think the guard on devices.empty() makes sense here anymore; if the user is req'd to specify a platform and device id, then it should fail via the new ( use_device_id >= devices.size() ) check, no fall though to the old use_devices.empty() check.